react/features/base/tracks/actions.js

```diff
@@ -89,9 +95,46 @@ export function createLocalTracksA(options = {}) {
                 /* firePermissionPromptIsShownEvent */ false,
                 store)
             .then(
-                localTracks => dispatch(_updateLocalTracks(localTracks)),
-                reason =>
-                    dispatch(_onCreateLocalTracksRejected(reason, device)));
+                localTracks => {
+                    // Because GUM is called for 1 device (which is actually
+                    // a media type 'audio','video', 'screen' etc.) we should
+                    // not get more than one JitsiTrack.
+                    if (localTracks.length !== 1) {
+                        throw new Error(
+                            'Expected exactly 1 track, but was '
+                                + `given ${localTracks.length} tracks`
+                                + `for device: ${device}.`);
+                    }
+
+                    if (gumProcess.canceled) {
+                        return _disposeTracks(localTracks)
+                                .then(dispatch(_trackCreateCanceled(device)));
+                    }
+
+                    return dispatch(trackAdded(localTracks[0]));
+                },
+                // eslint-disable-next-line prefer-arrow-callback
+                function(reason) {
+                    return gumProcess.canceled
+                        ? dispatch(_trackCreateCanceled(device))
+                        : dispatch(
+                            _onCreateLocalTracksRejected(reason, device));
```

Comment:
I'd rather we do the branching as late as possible and, thus, have a much common as possible:

`return dispatch(gumProcess.canceled ? _trackCreateCanceled(device) : _onCreateLocalTracksRejected(reason, device));`

(Of course, the above should be formatted properly but I'm writing it on GitHub so...)

Reply:
Will that actually change anything ? Isn't it that the argument will be resolved, before the return anyway ? I'm just asking whether it's about style or about the actual execution.

Reply:
It's about style.